### PR TITLE
add support for file_list shuffle each epoch and fix float learning rate bug

### PR DESCRIPTION
**Pull request by vslyu**
Add support for file_list shuffle each epoch
This is an example in yaml:
phase:
- name: phase1
  >model: "{workspace}/model.py" # user-defined model
  shuffle_filelist: True
  dataset_name: dataset_train # select dataset by name
  thread_num: 1

- name: phase2
  >model: "{workspace}/model.py" # user-defined model
  shuffle_filelist: True
  dataset_name: dataset_infer # select dataset by name
  thread_num: 1

Fix float learning rate bug
learning rate(supported) : float or string write such as 1e-3 ... 
This is an example in yaml:
- hyper_parameters:
   > optimizer:
        class: SGD
        learning_rate: 1e-3 / learning_rate: 0.001
        ...

### CODE REVIEW

File: core/model.py [hunk 1]
```diff
@@ -177,6 +177,12 @@ def optimizer(self):
         opt_name = envs.get_global_env("hyper_parameters.optimizer.class")
         opt_lr = envs.get_global_env(
             "hyper_parameters.optimizer.learning_rate")
+        if not isinstance(opt_lr, (float, Variable)):
+            try:
+                opt_lr = float(opt_lr)
+            except ValueError:
+                raise ValueError("learning_rate%s must be in float type" %
```
Comment (seiriosPlus) [inline core/model.py:184]: must be in float type ？

File: core/utils/util.py [hunk 2]
```diff
@@ -223,6 +224,18 @@ def check_filelist(hidden_file_list, data_file_list, train_data_path):
             return hidden_file_list, data_file_list
 
 
+def shuffle_files(model_dict, context):
```
Comment (seiriosPlus) [inline core/utils/util.py:227]: utils为啥可以拿到 context？

**Review by seiriosPlus — APPROVED** [review]

**Review by fuyinno4 — APPROVED** [review]

**Review by fuyinno4 — APPROVED** [review]

**Review by fuyinno4 — APPROVED** [review]

**Review by fuyinno4 — APPROVED** [review]
